scopus-import/src/main/java/no/sikt/nva/scopus/conversion/AffiliationGenerator.java

```diff
@@ -82,39 +75,47 @@ private List<CorporationWithContributors> fillCristinOrganizationData(
+        CorporationWithContributors corporation) {
+        return Optional.ofNullable(corporation.getScopusAuthors())
+                   .map(AuthorGroupTp::getAffiliation)
+                   .map(AffiliationTp::getCountry)
+                   .flatMap(this::searchCristinOrganization);
     }
 
-    private CorporationWithContributors searchForCristinOrganizationByName(CorporationWithContributors corporation) {
-        var organizationNames = extractOrganizationNames(corporation.getScopusAuthors());
-        var cristinOrganizations =
-            organizationNames.stream()
-                .map(this::searchCristinOrganization)
-                .flatMap(Optional::stream)
-                .toList();
-        return corporation.copy().withCristinCorporations(cristinOrganizations).build();
+    private Optional<CristinOrganization> searchForCristinOrganizationByName(CorporationWithContributors corporation) {
+        return extractOrganizationNames(corporation.getScopusAuthors()).stream()
+                   .map(this::searchCristinOrganization)
+                   .flatMap(Optional::stream)
+                   .findFirst();
     }
 
-    private CorporationWithContributors fetCristinOrganizationsById(CorporationWithContributors corporation) {
-        var organization = cristinConnection.fetchCristinOrganizationByCristinId(
-            corporation.getCristinOrganizationId());
-        List<CristinOrganization> validOrganizations = isValidOrganization(organization)
-                                                           ? List.of(organization)
-                                                           : List.of();
-        return corporation.copy().withCristinCorporations(validOrganizations).build();
+    private Optional<CristinOrganization> fetchCristinOrganizationsById(CorporationWithContributors corporation) {
+        return Optional.ofNullable(cristinConnection.fetchCristinOrganizationByCristinId(
+            corporation.getCristinOrganizationId()))
+                   .filter(this::isValidOrganization);
     }
 
     private boolean isValidOrganization(CristinOrganization organization) {
         return nonNull(organization) && isNotOtherNorwegianInstitution(organization);
     }
 
     private Optional<CristinOrganization> searchCristinOrganization(String organization) {
-        var searchResponse = cristinConnection.searchCristinOrganization(organization);
-        return searchResponse.isPresent() && hasSingleOrganization(searchResponse.get())
-                   ? Optional.ofNullable(getSingleOrganization(searchResponse.get()))
-                   : Optional.empty();
+        return cristinConnection.searchCristinOrganization(organization)
+                   .map(SearchOrganizationResponse::hits)
+                   .orElse(Collections.emptyList())
+                   .stream()
+                   .filter(cristinOrg -> cristinOrg.containsLabelWithValue(organization))
```

Comment:
This filter will also affect `searchForCristinOrganizationByName`, should the containsLabelWithValue()-method ignore case when comparing strings?

Reply:
It is the same for organization, we can fetch multiple items from Cristin when for example searching for NTNU, like NTNU, National Taiwan Normal University and NTNU Samfunnsforskning AS. We filtering out organizations that does not match organization name.